Adding keyboard shortcuts to Settings page

cedricium opened the pull request:
## Description:
This is the initial pass at adding keyboard shortcuts to the Settings page (see #276). Thus far, a `ShortcutsSection` class has been created and is applied to a
newly-added 'Keyboard Shortcuts' nav item. The template for ShortcutsSection
is essentially multiple `<div class="settings-cards">` elements containing one table of keyboard shortcuts organized by their underlying functionality (see attached screenshot).

![screen shot 2017-09-09 at 6 46 00 pm 2](https://user-images.githubusercontent.com/16343560/30245580-3d96430e-9593-11e7-9359-518a95a613a8.png)

The HTML `<kbd>` tag was defined in preference.css which styles the element to
look like a keyboard key.

## What's Next:
- [x] Add all keyboard shortcuts, currently only 3 categories have been added. (**updated 9/09** - 352b775)
)
- [x] Get feedback on design/layout, make any necessary changes

cedricium committed:
Adding keyboard shortcuts in Settings

This is the initial pass at adding keyboard shortcuts to the Settings page. In
this commit, the `ShortcutsSection` class has been created and is applied to a
newly-added 'Keyboard Shortcuts' nav item. The template for ShortcutsSection
is essentially multiple settings cards containing one table of keyboard short-
cuts organized by their underlying functionality.

The HTML `<kbd>` tag was defined in preference.css which styles the element to
look like a keyboard key, similar to StackOverflow or GitHub.

cedricium committed:
Added all keyboard shortcuts

Finished adding all keyboard shortcuts to the Settings page. Styled the tables
such that they are uniform with their columns being the same
width.

At the bottom of the 'Keyboard Shortcuts' settings page, a link to the
complete keyboard shortcuts documentation
(https://chat.zulip.org/help/keyboard-shortcuts) was also added.

cedricium commented:
Here's what the 'Keyboard Shortcuts' settings page looks like:

![zulip_shortcuts_settings](https://user-images.githubusercontent.com/16343560/30246574-3dcd3b3c-95b3-11e7-88cf-01e12ad2b124.gif)

penCsharpener commented:
nice! One thought: not write `Ctrl/Cmd` but distinguish between platforms to write only the appropriate. I think that would add a little to cleanness and avoid potential line breaks like in the `send message` shortcut of the first version.

cedricium committed:
Changed 'Ctrl/Cmd' to appropriate user OS key

If Windows or Linux, variable `userOSKey` will be 'Ctrl'. For Macs, `userOSKey`
will be '⌘' and these values will show up in place of the previous 'Ctrl/Cmd' keys.

cedricium commented:
@vbNETonIce I was initially thinking of doing it that way, just went ahead and made the change. I agree it adds a level of cleanliness by having OS-specific keys :)

akashnimare commented:
You have to implement the application's shortcuts (not the web app) which are present in the menu items, i.e. -

* Application shortcut -

![image](https://user-images.githubusercontent.com/2263909/30249772-9e5f7692-9660-11e7-82b6-a10711b6f0e6.png)

* Edit Shortcuts -

![image](https://user-images.githubusercontent.com/2263909/30249778-b8c39266-9660-11e7-986a-378486ce1e2e.png)

* View Shortcuts -

![image](https://user-images.githubusercontent.com/2263909/30249780-c7e4de12-9660-11e7-9005-1f089b3d7bd3.png)

* History Shortcuts -

![image](https://user-images.githubusercontent.com/2263909/30249788-e03bb5da-9660-11e7-83c6-c032643aa874.png)

* Window Shortcuts -

![image](https://user-images.githubusercontent.com/2263909/30249795-f3fa1eea-9660-11e7-8ffb-2dec9cb6df54.png)

cedricium commented:
@akashnimare oops, should've asked before doing all of those other shortcuts. I've made the changes now, check it out when you get a chance.

cedricium committed:
List application-specific shortcuts only

akashnimare commented:
Looks good. Can you rename `Keyboard Shortcuts` to `Shortcuts`?

akashnimare commented on app/renderer/js/pages/preference/nav.js:
```diff
@@ -8,7 +8,7 @@ class PreferenceNav extends BaseComponent {
 		this.props = props;
 
-		this.navItems = ['General', 'Network', 'Servers'];
+		this.navItems = ['General', 'Network', 'Servers', 'Keyboard Shortcuts'];
```
Replace `Keyboard Shortcuts` to `Shortcuts`.

akashnimare commented on app/renderer/js/pages/preference/preference.js:
```diff
@@ -56,6 +57,12 @@ class PreferenceView extends BaseComponent {
 				});
 				break;
 			}
+			case 'Keyboard Shortcuts': {
```
^^ Same.

cedricium committed:
Shorten 'Keyboard Shortcuts' to just 'Shortcuts'

cedricium commented:
@akashnimare glad you like it. Made the change you asked for. I'm going to go ahead and say this PR is no longer a WIP.

akashnimare commented:
Merged. Thanks, @cedricium 🚀
